docs/components/registrations.md

```diff
@@ -0,0 +1,138 @@
+# Registrations
+
+Components generally need to talk to one another!
+In simple cases, that occurs by method calls.
+But in many cases, a single component needs to communicate with a number of other components that all share some characteristics.
+For example, the `comp/core/health` component monitors the health of many other components, and `comp/workloadmeta/scheduler` provides workload events to an arbitrary number of subscribers.
+
+The convention in the Agent codebase is to use [value groups](./fx.md#value-groups) to accomplish this.
+The _collecting_ component requires a slice of some _collected type_, and the _providing_ components provide values of that type.
+Consider a simple case of an HTTP server component to which endpoints can be attached.
+The server is the collecting component, requiring a slice of type `[]*endpoint`, where `*endpoint` is the collected type.
+Providing components provide values of type `*endpoint`.
+
+The convention is to "wrap" the collected type in a struct type which embeds `fx.Out` and has tag `group:"pkgname"`, where `pkgname` is the short package name (Fx requires a group name, and this is as good as any).
+This helps providing components avoid the common mistake of omitting the tag.
+The collected type can be exported, such as if it has useful methods for providing components to call, but can also be unexported.
```

Comment:
```suggestion
The collected type can be exported, for example, if it has useful methods for providing components to call, but does not need to be exported.
```

Reply:
"unexported" is the term used for lower-case item names in Go, but this sentence is pretty confusing anyway.  Let me see what I can do :)